### ISSUE TITLE
Bug: 'SharedArrayBuffer' is not defined (using v^17.0.2)

### ISSUE THREAD

**Issue by baalimago** [facebook/react#21951]
## Issue:

It says `'SharedArrayBuffer' is not defined` when attempting to run project using a shared array buffer

## How to recreate:

1. `npx create-react-app test-project`
2. In App.js, add function `const test = () => { const b = new SharedArrayBuffer(1024); }`
3. Attempt to start app
4. Expect:
![image](https://user-images.githubusercontent.com/21121209/126866551-520287c6-3355-4f28-816b-0a7a45ef4129.png)

## System:
package.json:
```
{
  "name": "shared_array_buffer_test",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "^5.11.4",
    "@testing-library/react": "^11.1.0",
    "@testing-library/user-event": "^12.1.10",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-scripts": "4.0.3",
    "web-vitals": "^1.0.1"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}

```

Windows 11 (insider build 10.0.22000 Build 22000)
WSL2 - Ubuntu 20.04 LTS

**Comment by eps1lon** [comment]
Thanks for the feedback. 

This lint error originates from `react-scripts` which is released from [facebook/create-react-app](https://github.com/facebook/create-react-app). You want to [open a new issue in their repository](https://github.com/facebook/create-react-app/issues/new/choose)

**Comment by baalimago** [comment]
I don't think it's a lint issue.

It's caused by https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Opener-Policy and https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Embedder-Policy not being set properly in the react development server. It should've been fixed in react >=17.0.2 https://github.com/facebook/react/pull/20831, but it doesn't seem to work for some reason. 

I might add that it's not wsl2 related, I did the same test on ubuntu 20.04 and the behaviour is the same.

**Comment by siddharthroy12** [comment]
I'm having the same issue
